xlunch.c

```diff
@@ -2253,18 +2261,38 @@ void renderEntry(Imlib_Image buffer, char title[256], node_t * current, Cursor *
         imlib_context_set_image(buffer);
         int text_w;
         int text_h;
+        int extra_chars;
 
         const size_t osz = strlen(current->title);
         size_t sz = osz;
         imlib_context_set_font(font);
-        do
-        {
-            strncpyutf8(title,current->title,sz);
-            if(sz != osz)
-                strcat(title,"..");
-            imlib_get_text_size(title, &text_w, &text_h);
-            sz--;
-        } while(text_w > cell_width-(text_after ? (icon_size != 0 ? icon_padding*2 : icon_padding) + icon_size + text_padding : 2*text_padding) && sz>0);
+
+        if (maxtextlength >= 0) {
+          // get full title (--maxtextlength 0)
+          title = current->title;
+          extra_chars = 0;
+          // if value greater than zero, truncate text
+          if (maxtextlength > 0){
+            if (strlen(title) > maxtextlength) {
+              extra_chars = strlen(title)-maxtextlength;
```

Comment:
Since you're adding two dots after the title when truncated shouldn't you take those into account here? This risks making the text string longer if `strlen(title)` is only a single character longer than `maxtextlength`.

Reply:
Maybe.. I don't know C ..  But I give `--maxtextlength 10`, and I get 10 chars before the `..` ... and if I give `--maxtextlength 1`, and I get 1 char before the `..` - which is what I would expect...

Reply:
Ah, if that is what you expect then it's okay. I would think `maxtextlength` would mean that if my text is longer than `maxtextlength` it would cut it down to fit within `maxtextlength`. In this case setting `--maxtextlength 1` would mean every string would be converted to be three characters long.

May I ask what you're using this option for by the way?

Reply:
Ah okay, have you seen [this](https://github.com/Tomas-M/xlunch/blob/master/docs/spacings.svg) by the way? What you describe should be possible to achieve with large `iconpadding` and small `textpadding`.

Reply:
I had not ... WIll have a play with it..

Reply:
The problem with setting --iconpadding to high value (to get larger text) is that it will also affect vertical padding, setting e.g. --leastvmargin 1 doesn't help.
The option from sc0ttj --maxtextlength combined with high value of --leastmargin makes things more flexible for me.

Reply:
Well of course, you must allow more space for the text. I just tried @sc0ttj's solution and it has a couple of issues. Since no more space is required the text from one entry can overflow onto another entry, and the highlight around an icon might not cover all the text. I'm not sure if that's acceptable..

Reply:
> So instead of --maxtextlength you can now use --iconpadding <large number> --iconvpadding <10, or another low number>. 
The `--iconvpadding` is a great addition, I can confirm that it's working well, thanks !